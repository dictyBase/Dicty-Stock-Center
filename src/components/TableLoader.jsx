// @flow
import React, { Component } from 'react'
import { Grid, Cell } from 'radium-grid'
import 'styles/core.scss'

type Props = {
    title: string,
    message: string
}

export default class TableLoader extends Component {
    displayName = 'table loader'
    // static propTypes = {
    //     title: PropTypes.string,
    //     message: PropTypes.string
    // }
    render() {
        const title: string = this.props.title
        const message: string = this.props.message
        return (
            <div style={ {width: '100%'} }>
                <Grid cellWidth="1">
                    <Cell align="center">
                        <div className="text-center">
                            <h1>{ title && title }</h1>
                            <i className="fa fa-spinner fa-2x fa-pulse fa-fw margin-bottom"></i>
                            { message && message }
                        </div>
                    </Cell>
                </Grid>
            </div>
        )
    }
}
