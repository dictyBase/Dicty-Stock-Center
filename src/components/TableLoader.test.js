import React from "react"
import renderer from "react-test-renderer"
import TableLoader from "./TableLoader"

test("matching a snapshot of TableLoader", () => {
  const component = renderer.create(<TableLoader />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
