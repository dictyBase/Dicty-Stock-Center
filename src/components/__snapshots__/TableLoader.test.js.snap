// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of TableLoader 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="sc-bxivhb bUZGRr sc-htpNat gReFkJ sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="sc-fcdeBU kRrYpB sc-ifAKCX cFlEyZ sc-bdVaJa iHZvIS"
      is={null}
    >
      <h1 />
      <span
        aria-hidden={true}
        className="fa fa-spinner fa-2x fa-pulse fa-fw"
      />
    </div>
  </div>
</div>
`;
